AsyncDisplayKit/Private/ASStackUnpositionedLayout.mm

```diff
@@ -174,17 +178,13 @@ static inline CGFloat scaledFlexShrinkFactor(const ASStackUnpositionedItem &item
  @param flexFactorSum The sum of each item's flex factor as determined by the provided violation.
  @return A lambda capable of computing the flex grow adjustment, if any, for a particular item.
  */
-static std::function<CGFloat(const ASStackUnpositionedItem &, BOOL)> flexGrowAdjustment(const std::vector<ASStackUnpositionedItem> &items,
-                                                                                        const CGFloat violation,
-                                                                                        const CGFloat flexFactorSum)
+static std::function<CGFloat(const ASStackUnpositionedItem &)> flexGrowAdjustment(const std::vector<ASStackUnpositionedItem> &items,
+                                                                                  const CGFloat violation,
+                                                                                  const CGFloat flexFactorSum)
 {
-  const CGFloat violationPerFlexFactor = floorf(violation / flexFactorSum);
-  const CGFloat remainingViolation = violation - (violationPerFlexFactor * flexFactorSum);
   // To compute the flex grow adjustment distribute the violation proportionally based on each item's flex grow factor.
-  // If there happens to be a violation remaining make sure it is allocated to the first flexible child.
-  return [violationPerFlexFactor, remainingViolation](const ASStackUnpositionedItem &item, BOOL isFirstFlex) {
-    // Only apply the remaining violation for the first flexible child that has a flex grow factor.
-    return violationPerFlexFactor * item.child.style.flexGrow + (isFirstFlex && item.child.style.flexGrow > 0 ? remainingViolation : 0);
+  return [violation, flexFactorSum](const ASStackUnpositionedItem &item) {
+    return floorf(violation * (item.child.style.flexGrow / flexFactorSum));
```

Comment:
Use `std::floor` and friends throughout, so that we get type-generic math on different platforms.